Plots/Streamlines/NCL_stream_9.py

```diff
@@ -0,0 +1,166 @@
+# Make color map 
+
+colormap = colors.ListedColormap(['darkblue', 'mediumblue', 'blue', 'cornflowerblue', 'skyblue', 'aquamarine',
+        'lime', 'greenyellow', 'gold', 'orange', 'orangered', 'red', 'maroon'])
+
+colorbounds = np.arange(0, 56, 4)
+
+norm = mcolors.BoundaryNorm(colorbounds, colormap.N) #colors.BoundaryNorm(colorbounds, colormap.N)
+
+################################################################################
+
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds1 = xr.open_dataset(gdf.get('netcdf_files/U500storm.cdf'))
+ds2 = xr.open_dataset(gdf.get('netcdf_files/V500storm.cdf'))
+
+################################################################################
+# Set figure
+fig = plt.figure(figsize=(10, 10))
+
+# Create first subplot on figure for map
+ax = fig.add_axes([0,0,1,1], projection=ccrs.LambertAzimuthalEqualArea(central_longitude=-100, central_latitude=40), frameon=False, aspect='auto')
+
+#.1,.2,.8,.6
+
+# Set title of plot
+# Make title font bold using r"$\bf{_______}$" formatting
+gvutil.set_titles_and_labels(ax, maintitle=r"$\bf{Assigning}$"+" "+r"$\bf{color}$"+" "+r"$\bf{palette}$"+" "+r"$\bf{to}$"+" "+r"$\bf{streamlines}$", maintitlefontsize=25)
+
+# Set axis projection
+ax.set_extent((-128, -58, 18, 65), crs=ccrs.PlateCarree())
+
+# Add ocean, lakes, land features, and coastlines to map
+ax.add_feature(cfeature.OCEAN, color='lightblue')
+ax.add_feature(cfeature.LAKES, color='white', edgecolor='black')
+ax.add_feature(cfeature.LAND, color='tan')
+ax.coastlines()
+
+# Extract streamline data from initial timestep
+U = ds1.u.isel(timestep=0)
+V = ds2.v.isel(timestep=0)
+
+# Plot streamline data
+streams = ax.streamplot(U.lon, U.lat, U.data, V.data, transform=ccrs.PlateCarree(), arrowstyle='-', linewidth=3, density=2.0, color=U.data, cmap=colormap)
+
+# Divide streamlines into segments
+seg = streams.lines.get_segments()
+
+# Determine how many arrows on each streamline, the placement, and angles of the arrows
+period = 7
+arrow_x = np.array([seg[i][0, 0] for i in range(0, len(seg), period)])
+arrow_y = np.array([seg[i][0, 1] for i in range(0, len(seg), period)])
+arrow_dx = np.array([seg[i][1, 0] - seg[i][0, 0] for i in range(0, len(seg), period)])
+arrow_dy = np.array([seg[i][1, 1] - seg[i][0, 1] for i in range(0, len(seg), period)])
+
+#plt.scatter(arrow_x, arrow_y, zorder=12, color='pink')
+
+# Save figure to access color values of pixels
+plt.savefig('plot.png')
+im = Image.open(r"plot.png")
+
+plotsize = fig.get_window_extent().transformed(fig.dpi_scale_trans.inverted())
+plotwidth = (plotsize.width*fig.dpi)/2
+plotheight = (plotsize.height*fig.dpi)/2
+
+axsize = ax.get_window_extent().transformed(fig.dpi_scale_trans.inverted())
+axwidth = (axsize.width*fig.dpi)/2
+axheight = (axsize.height*fig.dpi)/2
+
+# Get x and y data, transform it into pixels, return RGB value of the pixels
+def getPixelVals(x, y):
```

Comment:
There are a lot of hard coded values here. Could you generalize this function or add comments about what these values represent?

Reply:
The hard coded values are the transform needed to find the location of the pixel. This solution is very roundabout! Because I am dividing each streamline into separate segments, and then plotting the arrows on those segments, I can't use the original 'U.data' argument used in the full streamline to determine the color. If you have a solution that doesn't involve manually finding the color of the pixels, that would be much appreciated haha.

Reply:
The color of the streamline is the magnitude of the wind right? If so, you can calculate the magnitude using the U and V components then use the 5th positional argument of `quiver` to specify that the magnitude is the value that will determine the color. Assign `cmap` to the same color map as the streamline and it should work.